.circleci/docker/build.sh

```diff
@@ -247,6 +270,47 @@ case "$image" in
     VISION=yes
     ROCM_VERSION=3.5.1
     ;;
+  *)
```

Comment:
I feel like this is rather a treatment of a symptom rather than an actual solution.

Like the idea of basing the properties of an image off of the name of said image always seemed like a weird concept to me.

Rather, we should just be explicitly passing the args directly through CircleCI environment variables rather than depending on the name to be within a specific format.

That being said this solution seems okay but brittle in that it relies on magic strings.

Curious to hear your thoughts on this as well @malfet

Reply:
I agree with @seemethere that original design is rather bad, but change seems ok if it can be used to pattern-match number of existing cases (which can be deleted as result).
Also, pattern:
```
  if [[ "$image" == *gcc* ]]; then
      GCC_VERSION="$(echo "${image}" | perl -n -e'/gcc(\d+(\.\d+)?)/ && print $1')"
      check_variable GCC_VERSION
    fi
```
can be refactored into something like
```
extract_version_from_image_name gcc GCC_VERSION
```

Reply:
fwiw, the code that currently runs currently does this hybrid "some of the docker properties are based off of naming, but other parts are just hardcoded", since we have a big switch statement on image names which toggle properties. So I think the code here is actually closer to what @seemethere  described as a better state than you might think. There is a sort of chesterton style reason for this, which is that you need to get someone to run the terraform to add a new image name, so people try to avoid doing that as much as possible. 

And weighing in as an old guard, IDM either way you guys go. Just explaining some of the third rails.

Reply:
I've added functions to extract all fields from the image name, but it's not currently used anywhere.  Please review latest commit(s) whether we want to use it.